app/modules/gpio.c

```diff
@@ -32,8 +33,8 @@ static void gpio_intr_callback_task (task_param_t param, uint8 priority)
     // GPIO callbacks are run in L0 and inlcude the level as a parameter
     lua_State *L = lua_getstate();
     NODE_DBG("Calling: %08x\n", gpio_cb_ref[pin]);
-    // The trigger is reset before the callback, as the callback itself might reset the trigger
-    pin_trigger[pin] = true;
+    // Reenable the interrupt just before calling the callback
+    platform_gpio_intr_init(pin, pin_int_type[pin]);
```

Comment:
The  original code enabled the interrupt _after_ the `lua_call` and the ISR has already disabled interrupts and cleared down the interrupt state, so we only need to do a `gpio_pin_intr_state_set(GPIO_ID_PIN(pin_num[pin]), type);` to re-enable interrupts.  The issue is _when_ from a temporal logic POV.  

-  In the case of a level interrupt the Lua cb code carries out the I/O actions to remove the level, so we should do it after the call otherwise we will get another "echo" task being posted.
-  In the case of an edge interrupt, doing it before isn't going to create the echo trigger, but there are going to be a few 100 instruction times (say 2-10µSec) where the last interrupt has been posted and we could miss an edge before reenabling the interrupts.